Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -925,28 +925,116 @@ extension SwiftLanguageServer {
     }
   }
 
+  func retrieveRefactorCodeActions(_ params: CodeActionRequest, completion: @escaping CodeActionProviderCompletion) {
+    guard let snapshot = documentManager.latestSnapshot(params.textDocument.url) else {
+      log("failed to find snapshot for url \(params.textDocument.url)")
+      completion([])
```

Comment:
Should return an error.

Reply:
Returning an error would work now since we only have one provider for code actions, but for the future the reasoning was that I didn't wanted to fail the entire request if a single provider failed. Should we go for it anyway?

Reply:
I added support for the providers to return an error but kept the code action request itself always succeeding. We can make it fail easily now if necessary.